domain/application/state/application.go

```diff
@@ -65,15 +69,25 @@ func (st *State) GetModelType(ctx context.Context) (coremodel.ModelType, error)
 // [applicationerrors.ApplicationAlreadyExists] if the application already exists.
 // It returns as error satisfying [applicationerrors.CharmNotFound] if the charm
 // for the application is not found.
-func (st *State) CreateApplication(ctx domain.AtomicContext, name string, args application.AddApplicationArg) (coreapplication.ID, error) {
+func (st *State) CreateApplication(
+	ctx context.Context,
+	name string,
+	args application.AddApplicationArg,
+	units ...application.AddUnitArg,
```

Comment:
I'm wondering in a future pass, just to add these to the AddApplicationArg.

At any rate, this should be a slice of unit args, and not an overloaded variadic functions.